runtime/vm/createramclass.cpp

```diff
@@ -2312,13 +2312,8 @@ internalCreateRAMClassDone(J9VMThread *vmThread, J9ClassLoader *classLoader, J9C
 				}
 			}
 #if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
-			if (J9ROMCLASS_IS_PRIMITIVE_VALUE_TYPE(romClass)
-				|| J9_ARE_ALL_BITS_SET(classFlags, J9ClassAllowsInitialDefaultValue)
-			) {
+			if (J9_ARE_ALL_BITS_SET(classFlags, J9ClassAllowsInitialDefaultValue)) {
 				UDATA instanceSize = state->ramClass->totalInstanceSize;
-				if (J9ROMCLASS_IS_PRIMITIVE_VALUE_TYPE(romClass)) {
-					classFlags |= J9ClassIsPrimitiveValueType;
```

Comment:
There are still many places checking this flag via `J9_IS_J9CLASS_PRIMITIVE_VALUETYPE()`.

Reply:
I was thinking of waiting to remove J9_IS_J9CLASS_PRIMITIVE_VALUETYPE since there are some uses of it for array support and I haven't yet implemented that for null restricted classes.

Reply:
Probably we can just remove the `J9AccPrimitiveValueType` and `J9ROMCLASS_IS_PRIMITIVE_VALUE_TYPE` in this PR. We can leave the changes regarding `J9ClassIsPrimitiveValueType` to a future PR.

Reply:
Okay. I removed the changes regarding J9ClassIsPrimitiveValueType except in createramclass.cpp.